yamale/validators/validators.py

```diff
@@ -166,6 +167,40 @@ def get_name(self):
         return self.regex_name or self.tag + " match"
 
 
+class Ip(Validator):
+    """IP address validator"""
+    tag = 'ip'
+    constraints = [con.IpVersion]
+
+    def _is_valid(self, value):
+        return self.ip_address(value)
+
+    def py2_ip_address(self, value):
+        try:
+            ipaddress.ip_interface(unicode(value))
+        except Exception:
```

Comment:
In Python exceptions are quite slow and should only be used for exceptional cases, not normal control flow. 

I'd suggest adding an `to_unicode` function in `utils.py`, next to the `isstr`. For Python 3 it would just return `str(input)` for Python 2: `unicode(input)`

Reply:
This comment isn't relevant to this exception BTW, but the one on line 188. Although there might be a better, non-exception way to do this one too.

Reply:
I know it's a point of debate and contention, but a frequently cited rule for writing Pythonic code is EAFP, which typically seems to result in try/except instead of `if`s.  Assume the best (and default), and handle the cases where it isn't that.  I can definitely make the requested change, just explaining my logic for doing it this way.  I'll work on it this week.

Reply:
Yes, it is Pythonic to ask for forgiveness, however, that implies you did something wrong in the first place :)

In any case, you're still following that principle, just in a place that's less of a performance impact.